src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/PositionExtensions.cs

```diff
@@ -22,6 +23,11 @@ public static int GetAbsoluteIndex(this Position position, SourceText sourceText
             }
 
             var linePosition = new LinePosition((int)position.Line, (int)position.Character);
+            if (linePosition.Line >= sourceText.Lines.Count)
+            {
+                Debug.Fail("Trying to get an absoute index beyond the sourceText, this might happen if your document is out of date.");
```

Comment:
If we're debug.failing it means we don't expect this ever to happen. So this can happeN?

Reply:
I saw this happen once when I removed a line, but I had also modified a lot of other random spots and wasn't able to get it to reproduce so there could have been complicating factors.

Reply:
Let's throw if this is called with a `SourceText` that doesn't match the input and catch bugs early. Given that this is an extension method, anyone could be calling this.